polkadot/node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -2107,6 +2119,106 @@ async fn handle_collation_fetch_response(
 	result
 }
 
+// Returns the claim queue without fetched or pending advertisement. The resulting `Vec` keeps the
+// order in the claim queue so the earlier an element is located in the `Vec` the higher its
+// priority is.
+fn unfulfilled_claim_queue_entries(relay_parent: &Hash, state: &State) -> Result<Vec<ParaId>> {
```

Comment:
Same here: I've added a todo in https://github.com/paritytech/polkadot-sdk/issues/6679 to keep `ClaimqueueState` in `State` and have it ready for use when we needed instead of rebuilding it each time.